waiter/src/waiter/token_watch.clj

```diff
@@ -7,12 +7,15 @@
             [metrics.timers :as timers]
             [waiter.correlation-id :as cid]
             [waiter.metrics :as metrics]
-            [waiter.token :as token]))
+            [waiter.token :as token]
+            [waiter.util.utils :as utils]))
 
 (defn make-index-event
   "Create an event for watch endpoints"
-  [type object]
-  {:object object :type type})
+  [type object & {:keys [id]}]
+  (cond->
+    {:object object :type type}
+    id (assoc :id id)))
```

Comment:
Every event should have an id, so this should not be conditional.

Reply:
For aggregate events (e.g. on refresh), this may be repetitive because all the ids will be the same. The current state only has top level events having an id guaranteed (INITIAL and EVENTS)

Reply:
Okay.